api/base/views.py

```diff
@@ -1,11 +1,65 @@
 from django.http import JsonResponse
 from rest_framework.decorators import api_view
 from rest_framework.response import Response
+from rest_framework import generics
 
 from api.users.serializers import UserSerializer
 from website import settings
 from .utils import absolute_reverse
 
+class JSONAPIBaseView(generics.GenericAPIView):
+
+    def __init__(self, **kwargs):
+        assert getattr(self, 'view_name', None), 'Must specify view_name on view.'
+        assert getattr(self, 'view_category', None), 'Must specify view_category on view.'
+        self.view_fqn = ':'.join([self.view_category, self.view_name])
```

Comment:
This was my suggestion.  `view_fqn` == view_fully_qualified_name.   The reverse resolver need the fqn, i.e. 'node:node-contributors', but the router just needs the view_name 'node-contributors'.  Rather than duplicating it, I suggested she put it in the init.